packages/core/admin/admin/src/content-manager/history/pages/History.tsx

```diff
@@ -1,27 +1,71 @@
+import * as React from 'react';
+
 import { Flex } from '@strapi/design-system';
-import { LoadingIndicatorPage } from '@strapi/helper-plugin';
+import { LoadingIndicatorPage, useQueryParams } from '@strapi/helper-plugin';
+import { stringify } from 'qs';
 import { Helmet } from 'react-helmet';
 import { useIntl } from 'react-intl';
-import { useParams } from 'react-router-dom';
+import { Navigate, useNavigate, useParams } from 'react-router-dom';
 
 import { useContentTypeLayout } from '../../hooks/useLayouts';
+import { buildValidGetParams } from '../../utils/api';
 import { VersionDetails } from '../components/VersionDetails';
 import { VersionsList } from '../components/VersionsList';
+import { useGetHistoryVersionsQuery } from '../services/historyVersion';
+
+import type { UID } from '@strapi/types';
 
 const HistoryPage = () => {
   const { formatMessage } = useIntl();
-  const { slug } = useParams<{
-    collectionType: string;
-    singleType: string;
-    slug: string;
+  const navigate = useNavigate();
+  const { slug, id: documentId } = useParams<{
+    slug: UID.ContentType;
+    id: string;
   }>();
 
-  const { isLoading, layout } = useContentTypeLayout(slug);
+  const { isLoading: isLoadingLayout, layout } = useContentTypeLayout(slug);
+
+  // Parse state from query params
+  const [{ query }] = useQueryParams<{
+    page?: number;
+    id?: string;
+    plugins?: Record<string, unknown>;
+  }>();
+  const validQueryParams = buildValidGetParams(query);
+  const page = validQueryParams.page ? Number(validQueryParams.page) : 1;
+
+  const versionsResponse = useGetHistoryVersionsQuery({
+    contentType: slug!,
+    ...(documentId ? { documentId } : {}),
+    ...validQueryParams,
+  });
+
+  // Make sure the user lands on a selected history version
+  React.useEffect(() => {
+    const versions = versionsResponse.data?.data;
+
+    if (!versionsResponse.isLoading && !query.id && versions?.[0]) {
```

Comment:
do you actually need to know if it's loading? surely you can just check against the data and if it's defined or not?